AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -141,29 +141,29 @@ + (BOOL)openURL:(NSURL *)url {
 }
 
 + (void)login {
-
-  // TODO protect with canBeUsed.
-  [[MSIdentity sharedInstance] login];
+  @synchronized(lock) {
+    if ([[MSIdentity sharedInstance] canBeUsed]) {
+      [[MSIdentity sharedInstance] login];
+    }
+  }
 }
 
 - (void)login {
-  @synchronized(lock) {
-    if (self.clientApplication == nil && self.identityConfig == nil) {
-      self.loginDelayed = YES;
-      return;
-    }
-    self.loginDelayed = NO;
-    [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
-                                  completionBlock:^(MSALResult *result, NSError *e) {
-                                    // TODO: synchronize accessToken assignment if it has threading issues
-                                    if (e) {
-                                      MSLogError([MSIdentity logTag], @"Couldn't initialize authentication client. Error: %@", e);
-                                    } else {
-                                      NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
-                                      self.accessToken = result.accessToken;
-                                    }
-                                  }];
+  if (self.clientApplication == nil || self.identityConfig == nil) {
+    self.loginDelayed = YES;
+    return;
   }
+  self.loginDelayed = NO;
+  [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                                completionBlock:^(MSALResult *result, NSError *e) {
+                                  // TODO: synchronize accessToken assignment if it has threading issues
+                                  if (e) {
+                                    MSLogError([MSIdentity logTag], @"Couldn't initialize authentication client. Error: %@", e);
+                                  } else {
+                                    NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
+                                    self.accessToken = result.accessToken;
```

Comment:
I am under the impression that we want to use `idToken` instead of `accessToken`. `idToken` allows you to identify a user while `accessToken` grants you the permission to a resource. Can you explain why we're using `accessToken`?

Reply:
This isn't related to this PR. It can be changed if necessary. The initial implementation was introduced when we spiked the feature, at that time it was just an example and we haven't considered this much at the moment.
However this PR is an actual implementation of login method, it should be changed in the PR if required.
So, I guess @dhei's comment is turning to rename the variable?

Reply:
I double checked the [docs for id token](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1302#discussion_r257066415):
> ID tokens are represented as JWTs, and they contain claims that you can use to identify users in your app. 

We should use id token instead of access token here, and Android SDK already does that at https://github.com/Microsoft/AppCenter-SDK-Android/blob/feature/identity/sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java#L379-L380.

Reply:
Thanks for clarifying. I'll fix this.

Reply:
Yep we should use ID token. Android is already using ID tokens

Reply:
`accessToken` was renamed to `idToken` and `login` method was updated. Please take a look.